### PR TITLE
Fix state dir in docker mode

### PR DESCRIPTION
**Pull request by rbren**
**End-user friendly description of the problem this fixes or functionality that this introduces**

- [ ] Include this change in the Release Notes. If checked, you must provide an **end-user friendly** description for your change below
no changelog
---
**Give a summary of what the PR does, explaining any non-trivial design decisions**

Looks like there's a hard failure if you don't mount the volume

---
**Link of any specific issues this addresses**

---

To run this PR locally, use the following command:
```
docker run -it --rm   -p 3000:3000   -v /var/run/docker.sock:/var/run/docker.sock   --add-host host.docker.internal:host-gateway   -e SANDBOX_RUNTIME_CONTAINER_IMAGE=docker.all-hands.dev/all-hands-ai/runtime:5471930-nikolaik   --name openhands-app-5471930   docker.all-hands.dev/all-hands-ai/openhands:5471930
```

### CODE REVIEW

**Review by enyst — APPROVED** [review]

**Review by mamoodi — APPROVED** [review]

**Review by tofarr — APPROVED** [review]
🍰

**Comment by mamoodi** [comment]
Just to understand, with this change we are expecting the following to be persisted now inside the folder
* JWT token
* all your historical conversations
* your LLM settings

Is that correct?

**Comment by rbren** [comment]
correct! Will test once the docker image pops out

**Comment by rbren** [comment]
OK a few fixes in here:
* runs correctly when volume is not mounted
* mounting correctly persists things to disk
* we now migrate to the backend version of settings on startup

File: frontend/src/services/settings.ts [hunk 1]
```diff
@@ -68,6 +114,11 @@ export const maybeMigrateSettings = (logout: () => void) => {
   if (currentVersion < 4) {
     logout();
   }
+
+  if (currentVersion < 5) {
+    const localSettings = getLocalStorageSettings();
+    saveSettings(localSettings);
```
Comment (rbren) [inline frontend/src/services/settings.ts:120]:
```suggestion
    await saveSettings(localSettings);
```

**Pull request merged**